lib/tapjoy/ldap/user/create.rb

```diff
@@ -0,0 +1,83 @@
+require 'digest'
+require 'securerandom'
+module Tapjoy
+  module LDAP
+    module User
+      class Create
+        def opts
+          @opts ||= Trollop::options do
+            # Set help message
+            usage 'user create [options]'
+            synopsis "\nThis command is for creating new LDAP users"
+
+            # Username is two arguments
+            # Trollop will accept more, but we will only parse two later
+            # TODO: support given names that include a space
+            opt :user, "Specify user's first and last name", type: :strings, required: true
+
+            # Groupname is a single string, for primary group setting
+            opt :group, 'Specify name of primary group', type: :string, required: true
+            opt :type, 'Specfy if this is a user or service account', type: :string, default: 'user'
+          end
+        end
+
+        def uidnumber
+          @uidnumber ||= Tapjoy::LDAP::client.get_max_id('user', opts[:type])
+        end
+
+        def gidnumber
+          @gidnumber ||= Tapjoy::LDAP::Group.lookup_id(opts[:group])
+        end
+        
+        def create
+          # Check for errors
+          Trollop::die :user, 'argument count must be two' if opts[:user].size != 2
+          Trollop::die :type, "argument must be 'user' or 'service'" unless ['user', 'service'].include?opts[:type]
+
+
+          case opts[:type]
+          when 'user'
+            ou = 'People'
+          when 'service'
+            ou = Tapjoy::LDAP::client.service_ou
+          else
+            puts 'Unknown type'
+          end
+
+          dn = "uid=#{ username },ou=People,#{ Tapjoy::LDAP::client.basedn }"
```

Comment:
based on the case statement above it looks like `ou=People` should be `ou=#{ou}`